### PR TITLE
Various terminal quick fix fixes

### LINKED ISSUE #161907: Terminal quick fix title should show the command that should be run
Testing #161846

1. `git checkout -b lalala`
2. `git push`
3. Notice quick fix title is "git push lalala". Title should be what will actually be run `git push --set-upstream origin lalala`

Right now the user does not know what will be run, so it is surprising and does not give confidence.

Screenshot 2022-09-27 at 10 20 01

### LINKED ISSUE #162070: Terminal quick fixes end up matching against the text of all rows at the end of the output matcher
It seemed a bit off why the renderer froze for 50 seconds because of this when scanning a command's output and this is why:

image

We should be doing `line.match`, not `output.match` above

### LINKED ISSUE #162074: Create PR terminal quick fix isn't working on Windows
Clicking it was navigating to `https://github.com/microsoft/vscode/compare/tyriar/wording%20%20%20%20%20%20%20%20remote:To%20https:/github.com/microsoft/vscode%20*%20%5Bnew%20branch%5D%20%20%20%20%20%20%20%20%20%20%20%20%20%20tyriar/wording%20-%3E%20tyriar/wordingBranch%20'tyriar/wording'%20set%20up%20to%20track%20remote%20branch%20'tyriar/wording'%20from%20'origin'.?expand=1` for me

### PR DESCRIPTION
**Pull request by Tyriar**
Fixes https://github.com/microsoft/vscode/issues/162070
Fixes https://github.com/microsoft/vscode/issues/162074
Fixes https://github.com/microsoft/vscode/issues/161907

---

Tested on Windows only, this moved create PR to a single line match

image
image

### CODE REVIEW

File: src/vs/workbench/contrib/terminal/browser/terminalQuickFixBuiltinActions.ts [hunk 1]
```diff
@@ -122,13 +121,12 @@ export function gitCreatePr(openerService: IOpenerService): ITerminalQuickFixOpt
...
-			const branch = matchResult?.outputMatch?.[1];
-			const link = matchResult?.outputMatch?.[2];
-			if (!branch || !link) {
+			const link = matchResult?.outputMatch?.[1];
+			if (!link) {
 				return;
 			}
 			const actions: IAction[] = [];
-			const label = localize("terminal.gitCreatePr", "Create PR");
+			const label = localize("terminal.openLink", "Open link: {0}", link);
```
Comment (meganrogge) [inline src/vs/workbench/contrib/terminal/browser/terminalQuickFixBuiltinActions.ts:129]: This is a bit vague - what about `Open PR link on GitHub`?

**Comment by meganrogge** [comment]
an issue with the single line approach:

Screen Shot 2022-09-27 at 4 26 04 PM

**Comment by meganrogge** [comment]
also no match found here because multi-line
Screen Shot 2022-09-27 at 4 27 29 PM

**Review by meganrogge — CHANGES_REQUESTED** [review]
my initial approach was similar to this, but realized there are issues with it - especially when dealing with split terminals

File: src/vs/workbench/contrib/terminal/browser/terminalQuickFixBuiltinActions.ts (hunk 1 above)
Comment (Tyriar) [reply src/vs/workbench/contrib/terminal/browser/terminalQuickFixBuiltinActions.ts:129]: Personally I think the URL is more transparent/useful than, especially early on when there's less trust in the feature; I'm scared of more bugs like https://github.com/microsoft/vscode/issues/162074 slipping in. Shall we keep it as open link for now refine in future iterations? Best of both worlds work be the new quick fix widget with the ability to show the URL/more info as well. Showing what free port in a details part would be nice too

Comment (meganrogge) [reply src/vs/workbench/contrib/terminal/browser/terminalQuickFixBuiltinActions.ts:129]: Yes that works for me

**Comment by Tyriar** [comment]
For those multi-line ones, the lines should be marked as wrapped so we should be able to fetch the the wrapped line. Will look into fixing that

**Comment by Tyriar** [comment]
Can repro and I see the issue:

Screen Shot 2022-09-27 at 5 00 39 pm

**Comment by Tyriar** [comment]
Fixed:

Screen Shot 2022-09-27 at 5 10 13 pm

Tests are probably not passing which I'll fix tomorrow, I also need to verify `anchor: 'top'`

**Review by meganrogge — APPROVED** [review]

**Pull request merged**